### DISCUSSION TITLE
Can verdaccio run as a npm proxy cache?

### DISCUSSION THREAD

**Discussion by patgmiller** [verdaccio/verdaccio#2319] [Q&A]
Previously I deployed verdaccio for a company's build cache because npm / yarn cloudflare was rate limiting our builds. However, after several hours of testing / tweaking it appears as though verdaccio does not support the npm / yarn proxy configurations, it only works when setting the `--registry` config to the verdaccio host, which resolves as though the verdaccio host IS the registry instead of just a proxy. Which means the resulting lock files will have the host domain and protocol 

So I am wondering if it is possible to setup verdaccio to handle `--proxy` install but respond with upstream registry?

**Comment by juanpicado** [comment]
🤔 I am not sure if I understand your question, could you provide detailed examples.

**Reply by patgmiller** [reply]
when using `verdaccio` it injects itself as the registry for public packages, even though `verdaccio` is not the official registry for a given package.

**Reply by patgmiller** [reply]
What I am asking is I want to be able to use it as the proxy server for 
```bash
yarn install --proxy "verdaccio.myhost.tld:port"
```

**Reply by patgmiller** [reply]
The docs for `verdaccio` indicate one should setup their node project with `--registry https://verdaccio.myhost.tld:port`. However with that setup any `npm` or `yarn` lock files will include that domain in the details. Now that good for packages I want to publish to my own registry, but I would not prefer that be the case for other public packages which use `registry.npmjs.org` or the yarn cname domain for that. 

Does that help clarify my question? Or do you still need other example details?

**Reply by juanpicado** [reply]
> However with that setup any npm or yarn lock files will include that domain in the details. 

If you are using npm (v6,v7) or yarn 1, yes, the `resolved` field will be on the lock file, this is inkonvenient as you says. But you have options, pnpm 6 or yarn 2 does not have this limitation.

> verdaccio does not support the npm / yarn proxy configurations

I'm not familiar with those setups, cannot help here.
